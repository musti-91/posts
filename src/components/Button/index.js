import React, { PureComponent } from "react";
import PropTypes from "prop-types";
//import { Test } from './Button.styles';

class Button extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      hasError: false
    };
  }

  componentWillMount = () => {
    console.log("Button will mount");
  };

  componentDidMount = () => {
    console.log("Button mounted");
  };

  componentWillReceiveProps = nextProps => {
    console.log("Button will receive props", nextProps);
  };

  componentWillUpdate = (nextProps, nextState) => {
    console.log("Button will update", nextProps, nextState);
  };

  componentDidUpdate = () => {
    console.log("Button did update");
  };

  componentWillUnmount = () => {
    console.log("Button will unmount");
  };

  render() {
    if (this.state.hasError) {
      return <h1>Something went wrong.</h1>;
    }
    return <div className="ButtonWrapper">Test content</div>;
  }
}

Button.propTypes = {
  // bla: PropTypes.string,
};

Button.defaultProps = {
  // bla: 'test',
};

export default Button;
