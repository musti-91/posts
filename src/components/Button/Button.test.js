import React from "react";
import { shallow } from "enzyme";

import Button from "./index";

describe("Button Component", () => {
  test("should match snapshot", () => {
    expect.assertions(1);

    const wrapper = shallow(<Button />);

    expect(wrapper).toMatchSnapshot();
  });
});
