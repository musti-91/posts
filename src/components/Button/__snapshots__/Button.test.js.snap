// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshot 1`] = `
<div
  className="ButtonWrapper"
>
  Test content
</div>
`;
